examples/counter/test/sagas.js

```diff
@@ -25,5 +26,20 @@ test('incrementAsync Saga test', (t) => {
     'counter Saga must be done'
   )
 
+  const watcher = rootSaga()
+  const worker = takeEvery('INCREMENT_ASYNC', incrementAsync);
+
+  t.deepEqual(
+    watcher.next().value,
+    worker.next().value,
+    'rootSaga takes from every INCREMENT_ASYNC action'
+  )
+
+  t.equal(
+    watcher.return().done,
```

Comment:
In this specific case it does but in other sagas (with try/catch/finally) this would not be true. So in this case the test asserts that takeEvery is **not** wrapped in a finally block.

On another note, this is especially done to increase coverage. If you want to get even better coverage than this one would need to run a `saga.throw(err)` assertion as well. I left this off here though as a bit too verbose.